.circleci/config.yml

```
@@ -1,170 +1,279 @@
+          command: make e2e-tests
+
+  cleanup:
+    steps:
+      - run:
+          name: Cleanup
+          command: make clean
+
+  build_application:
+    steps:
+      - run:
+          name: Build application
+          command: npm run build
+
+  package_app:
+    steps:
+      - run:
+          name: Package app
+          command: make package
+
+  check_legal_compliance:
+    steps:
+      - run:
+          name: Check for legal compliance
+          command: make legal
+
+  deploy_gh_pages:
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v${CACHE_VERSION}-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v${CACHE_VERSION}-dependencies-
+      - install_node_dependencies
+      - build_application
+      - add_ssh_keys:
+          fingerprints:
+            - "9a:fe:61:f4:d6:c2:c1:58:75:c9:72:56:6e:53:fd:8b"
+      - run:
+          name: Deploy to GitHub Pages
+          command: |
+            cd build
+            git config --global user.email circleci@circleci.com
+            git config --global user.name CircleCIBot
+            git init
+            git add -A
+            git commit -m 'Demo build [ci skip]'
+            git remote add origin https://github.com/quantumblacklabs/kedro-viz.git
+            git push origin master:gh-pages --force
+
+  build:
+    description: Run build
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v${CACHE_VERSION}-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v${CACHE_VERSION}-dependencies-
+      - setup_python_env
+      - install_node_dependencies
+      - run_eslint
+      - run_javascript_tests
+      - test_lib_transpilation
+      - run_pylint_and_flake8
+      - run_secret_scan
+      - run_security_scan
+      - run_python_tests
+      - run_e2e_tests
+      - cleanup
+      - save_cache:
+          paths:
+            - node_modules
+          key: v${CACHE_VERSION}-dependencies-{{ checksum "package.json" }}
+      - build_application
+      - package_app
+      - check_legal_compliance
+
+  # Windows-related commands
+  win_setup_conda:
+    # Miniconda3 is pre-installed on the machine:
+    # https://circleci.com/docs/2.0/hello-world-windows/
+    description: Setup conda
+    steps:
+      - run:
+          name: Initialize conda
+          command: conda init powershell
+      - run:
+          name: Create 'kedro-viz' conda environment
+          command: |
+            conda create --name kedro-viz python=$env:CONDA_ENV_PY_VERSION -y
+
+  win_setup_requirements:
+    description: Install Kedro-Viz dependencies
+    steps:
+      - run:
+          name: Install Kedro-Viz dependencies
+          command: |
+            conda activate kedro-viz
+              pip install -r package/test_requirements.txt -U
```

```suggestion
            pip install -r package/test_requirements.txt -U
```
Don't we want to do install kedro@master here as well?

---

I looks redundant because `pip install` installs requirements from `requirements.txt` which includes kedro.

---

I think the intention behind that is we always want to test kedro-viz against latest kedro master so we discover incompatibility even before we release.